src/test/java/bio/terra/service/snapshotbuilder/utils/CriteriaQueryBuilderTest.java

```diff
@@ -101,7 +101,8 @@ void generateFilterForDomainCriteriaThrowsIfGivenUnknownDomain() {
   @ParameterizedTest
   @ArgumentsSource(QueryTestUtils.Contexts.class)
   void generateFilterIdentifiesDomainCriteria(SqlRenderContext context) {
-    SnapshotBuilderCriteria criteria = generateDomainCriteria();
+    SnapshotBuilderCriteria criteria =
+        generateDomainCriteria(10, 0); // Domain 10 = condition_occurrence
```

Comment:
Would the mapping values be fetched from the `dataset_snapshot_builder_settings.json`? I ask this because in my `dataset_snapshot_builder_settings.json`, condition_occurrence is mapped to `id` 19.

Reply:
Good question - this was confusing to me. In the test setup, we run the following line of code:

```
criteriaQueryBuilder = new CriteriaQueryBuilder(Person.TABLE_NAME, SnapshotBuilderTestData.SETTINGS);
```

As far as I can tell, the settings are set by `SnapshotBuilderTestData.SETTINGS`, which is set [here](https://github.com/DataBiosphere/jade-data-repo/blob/cdaa608986166fc9e0a41be71f4a14ed1dfd29a9/src/test/java/bio/terra/service/snapshotbuilder/SnapshotBuilderTestData.java#L83). Condition maps to [domain 10](https://github.com/DataBiosphere/jade-data-repo/blob/cdaa608986166fc9e0a41be71f4a14ed1dfd29a9/src/test/java/bio/terra/service/snapshotbuilder/SnapshotBuilderTestData.java#L88) and procedure maps to [domain 11](https://github.com/DataBiosphere/jade-data-repo/blob/cdaa608986166fc9e0a41be71f4a14ed1dfd29a9/src/test/java/bio/terra/service/snapshotbuilder/SnapshotBuilderTestData.java#L98). (At least this is my understanding!!)

Reply:
Thanks for the detailed response!